docs/USAGE.md

```diff
@@ -15,12 +15,11 @@ To build the image, run:
 
 The build process will take about 20 or 30 minutes, depending on the host computer.
 
-The build process defaults to cloning the `ros2.repos` file from [spaceros](https://github.com/space-ros/space-ros) repository.
-It looks for a branch with the same name as the current local branch; if it doesn't find one, it falls back to cloning from the main branch.
-For testing purposes, you can customize both the spaceros repository URL and the branch name by modifying arguments defined in the [Earthfile](./Earthfile).
-Example:
-```bash
-earthly +image --SPACEROS_REPO_URL="https://github.com/my-org/my-spaceros-fork.git" --SPACEROS_GIT_REF="my-branch-name"
+The build will use the local [ros2.repos](../ros2.repos) file, and will pick up any local modifications.
+The ros2 repos file can be regenerated or updated using the build target,
+
+```
+earthly build +repos-file
```

Comment:
Gosh this is a weird one... I could swear this was working before. Also, I have no idea how this was working before since the old generate repos script did the same thing with an empty `AMENT_PREFIX_PATH`. I did note that the older images have `1.22` installed, whereas now I'm installing `1.23`. However, this check has been there for years:

https://github.com/ros-infrastructure/rosinstall_generator/blob/master/src/rosinstall_generator/generator.py#L203

In any case, this *should* be set once we actually have built and sourced anything for ROS. However, in this first instance `AMENT_PREFIX_PATH` is blank, so `ROS_PACKAGE_PATH` is blank, so it barfs. To get around this I just set it to the [empty spaceros directory](https://github.com/space-ros/space-ros/pull/170/files#diff-abd0a56959a8f57df0c5789b1d884f8fa1f0856228e4e0d60bb528831fede990R117). @xfiderek does that change look ok to you?

This should work now.

Reply:
Bump: I would appreciate another set of eyes just to make sure this isn't missing something silly.

Reply:
I think the initial idea that I had was that we would run `generate-repos.sh` with `--exclude-installed false`. This argument defaults to true, because we would like this set for own downstream images and it is more convenient for users, but in base image we need to set it to false. I should have specified it more transparently, so that is for me. When we run it with this argument set to false, it should run smoothly and the modification to `export ROS_PACKAGE_PATH` line would not be required anymore.

Reply:
No worries! Ok that makes much more sense. I removed that commit and modified the build target in the Earthly file to set `--exclude-installed false`.